projectq/ops/_uniformly_controlled_rotation.py

```diff
@@ -0,0 +1,145 @@
+#   Copyright 2018 ProjectQ-Framework (www.projectq.ch)
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+import math
+
+from ._basics import ANGLE_PRECISION, ANGLE_TOLERANCE, BasicGate, NotMergeable
+
+
+class UniformlyControlledRy(BasicGate):
+    """
+    Uniformly controlled Ry gate as introduced in arXiv:quant-ph/0312218.
+
+    This is an n-qubit gate. There are n-1 control qubits and one target qubit.
+    This gate applies Ry(angles(k)) to the target qubit if the n-1 control
+    qubits are in the classical state k. As there are 2^(n-1) classical
+    states for the control qubits, this gate requires 2^(n-1) (potentially
+    different) angle parameters.
+
+    Example:
+        .. code-block:: python
+
+        controls = eng.allocate_qureg(2)
+        target = eng.allocate_qubit()
+        UniformlyControlledRy(angles=[0.1, 0.2, 0.3, 0.4]) | (controls, target)
+
+    Note:
+        The first quantum register contains the control qubits. When converting
+        the classical state k of the control qubits to an integer, we define
+        controls[0] to be the least significant (qu)bit.
```

Comment:
On second thoughts:
Which order is more natural?

Reply:
Since it's about integers, I'd also choose this ordering of bits.